src/pages/index.js

```diff
@@ -109,21 +181,97 @@ function closeModal(modal) {
   modal.removeEventListener("mousedown", handleOverLayClose);
 }
 
+function handleLike(evt, id) {
+  const isLiked = evt.target.classList.contains("card__like-btn_liked");
+
+  api
+    .handleLikeStatus(id, isLiked)
+    .then((card) => {
+      if (card) {
+        evt.target.classList.toggle("card__like-btn_liked");
+        localStorage.setItem(`card-${id}-isLiked`, !isLiked);
+      }
+    })
+    .catch(console.error);
+}
 function handleProfileFormSubmit(evt) {
   evt.preventDefault();
-  profileNameEl.textContent = nameInput.value;
-  profileDescriptionEl.textContent = descriptionInput.value;
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+
+  api
+    .editUserInfo({ name: nameInput.value, about: descriptionInput.value })
+    .then((data) => {
+      profileNameEl.textContent = data.name;
+      profileDescriptionEl.textContent = data.about;
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false);
+    });
+
   closeModal(editModal);
```

Comment:
`  closeModal(editModal);
`

This should be in line 206 in `.then`

You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**